metrics/sender.go

```diff
@@ -50,8 +49,10 @@ type appInfo struct {
 	Version string `json:"version"`
 }
 
-type natMappingFailContext struct {
-	errorMessage string
+type natMappingContext struct {
+	stage        string
+	successful   bool
+	errorMessage *string
```

Comment:
Why not keep it as `error` type?

Reply:
I had it as `error` first, but somebody though it's confusing (because type is a `string` and not an `error`), so I fixed a change request :D and now we're going back :D is it really that important?

Reply:
Well, with `error` you will not have to convert `err` to string to assign it :)